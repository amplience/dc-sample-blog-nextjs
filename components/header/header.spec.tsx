/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import Header from './header';

const mockUseRouter = jest.fn();
jest.mock('next/router', () => {
  return {
    useRouter: () => mockUseRouter()
  };
});

describe('Header', () => {
  test('renders blog header', async () => {
    mockUseRouter.mockImplementationOnce(() => {
      return { asPath: '' };
    });
    const props = {
      title: 'Blog Title'
    };
    const component = await renderer.create(<Header {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders blog header with preview vse query string link', async () => {
    mockUseRouter.mockImplementationOnce(() => {
      return { asPath: '?vse=test-vse.domain' };
    });
    const props = {
      title: 'Blog Title'
    };
    const component = await renderer.create(<Header {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
