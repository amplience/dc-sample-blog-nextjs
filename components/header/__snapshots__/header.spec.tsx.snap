// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders blog header 1`] = `
<section
  className="jsx-2398603546"
>
  <header
    className="jsx-2398603546"
  >
    <a
      aria-label=""
      className="jsx-3354872520"
      href="/"
    >
      <img
        alt="Amplience logo"
        className="jsx-2718936118 logo"
        src="/static/images/logo-white-100h.png"
        srcSet="/static/images/logo-white-50h.png 1x, /static/images/logo-white-100h.png 2x"
      />
    </a>
    <span
      className="jsx-2398603546"
    >
      Blog Title
    </span>
  </header>
</section>
`;

exports[`Header renders blog header with preview vse query string link 1`] = `
<section
  className="jsx-2398603546"
>
  <header
    className="jsx-2398603546"
  >
    <a
      aria-label=""
      className="jsx-3354872520"
      href="/?vse=test-vse.domain"
    >
      <img
        alt="Amplience logo"
        className="jsx-2718936118 logo"
        src="/static/images/logo-white-100h.png"
        srcSet="/static/images/logo-white-50h.png 1x, /static/images/logo-white-100h.png 2x"
      />
    </a>
    <span
      className="jsx-2398603546"
    >
      Blog Title
    </span>
  </header>
</section>
`;
